Lib/unittest/mock.py

```diff
@@ -1469,13 +1474,58 @@ def get_original(self):
             )
         return original, local
 
+    @property
+    def is_started(self):
+        return self._context is not None
+
+    @property
+    def is_local(self):
+        return self._context.is_local
+
+    @property
+    def target(self):
+        return self._context.target
+
+    @property
+    def temp_original(self):
+        return self._context.original
+
+    @is_local.setter
+    def is_local(self, value):
+        self._context = _PatchContext(
+            exit_stack=self._context.exit_stack,
+            is_local=value,
+            original=self._context.original,
+            target=self._context.target,
+        )
```

Comment:
Urgh, I forgot that you cannot change the value of namedtuples. Ok, my suggestion using namedtuples was wrong. To reduce memory footprint, we can use `__slots__` in a regular class instead like you had before. That way, we save an `from collections import namedtuple` as well and simplify the property's setter. WDYT? (again sorry for this bad suggestion).